src/script/convert_url_index.sh

```diff
@@ -60,6 +60,8 @@ $SPARK_HOME/bin/spark-submit \
     --conf spark.sql.hive.metastorePartitionPruning=true \
     --conf spark.hadoop.parquet.enable.summary-metadata=false \
     --conf spark.sql.parquet.outputTimestampType=TIMESTAMP_MILLIS \
+    --conf "spark.driver.userClassPathFirst=true" \
+    --conf "spark.executor.userClassPathFirst=true" \
```

Comment:
Hmhh. Might be safe because cc-index-table ships with very few dependencies given that Spark is "provided" and the AWS SDK shades all transitive dependencies. In general, I'd be very cautious with this [configuration option](https://spark.apache.org/docs/latest/configuration.html) still marked as "experimental": conflicts with transitive dependencies from the user jar and used/required by Spark may cause errors (typically related to serialization or inter-process communication) which are difficult to trace.

Reply:
I could not get the job to run at all without these lines - the spark environment (as seen in the dockerfile) was overriding our libraries, and it was causing this error:

```
Caused by: java.lang.NoSuchMethodError: 'com.google.gson.JsonElement com.google.gson.JsonParser.parseReader(java.io.Reader)'
```

If you have documentation or hints you can give on how the real environment is setup, I'd appreciate a pointer, I'm happy to adjust.

Reply:
I know, I've also run into the gson issue. The version used must be the same (or API-compatible) than that provided by Spark. As said, I'm mostly optimistic that there will be no regressions with the user class path first. If not (to be tested), we can roll this back later.

Alternatively, the script already allows to add additional Spark options via the environment variable `SPARK_EXTRA_OPTS`:
```
SPARK_EXTRA_OPTS="--conf spark.driver.userClassPathFirst=true --conf spark.executor.userClassPathFirst=true" ./src/script/convert_url_index.sh ...
```

Reply:
The setup of Spark and the job configuration is in the crawl-tools repository. Ping me if more information is needed.